app/uk/gov/hmrc/calculatepublicpensionadjustment/services/PaacService.scala

```diff
@@ -0,0 +1,166 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.calculatepublicpensionadjustment.services
+
+import com.google.inject.Inject
+import uk.gov.hmrc.calculatepublicpensionadjustment.connectors.PaacConnector
+import uk.gov.hmrc.calculatepublicpensionadjustment.logging.Logging
+import uk.gov.hmrc.calculatepublicpensionadjustment.models.calculation.CalculationRequest
+import uk.gov.hmrc.calculatepublicpensionadjustment.models.calculation.cppa._
+import uk.gov.hmrc.calculatepublicpensionadjustment.models.calculation.paac._
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PaacService @Inject() (connector: PaacConnector)(implicit ec: ExecutionContext) extends Logging {
+
+  def calculate(
+    calculationRequest: CalculationRequest
+  )(implicit hc: HeaderCarrier): Future[Unit] =
+    for {
+      paacResponse <- sendRequest(calculationRequest)
+    } yield Future.unit
+
+  def sendRequest(
+    calculationRequest: CalculationRequest
+  )(implicit hc: HeaderCarrier): Future[PaacResponse] =
+    connector
+      .sendRequest(buildPaacRequest(calculationRequest))
+      .map { response =>
+        PaacResponse(response.rows flatMap { row =>
+          row.taxYear match {
+            case _: PaacTaxYear2011To2015.NoInputTaxYear | _: PaacTaxYear2016PreAlignment.NoInputTaxYear |
+                _: PaacTaxYear2016PostAlignment.NoInputTaxYear | _: PaacTaxYear2017ToCurrent.NoInputTaxYear =>
+              None
+            case _ => Some(row)
+          }
+        })
+      }
+
+  def buildPaacRequest(calculationRequest: CalculationRequest): PaacRequest = {
+    val paacTaxYears = calculationRequest.taxYears.map {
+      case CppaTaxYear2013To2015(pensionInputAmount, period) =>
+        PaacTaxYear2011To2015.NormalTaxYear(pensionInputAmount, period)
+
+      case CppaTaxYear2016PreAlignment.NormalTaxYear(pensionInputAmount, taxYearSchemes, _, _, period) =>
+        PaacTaxYear2016PreAlignment.NormalTaxYear(
+          pensionInputAmount + taxYearSchemes.map(_.revisedPensionInputAmount).sum,
+          period
+        )
+
+      case CppaTaxYear2016PreAlignment.InitialFlexiblyAccessedTaxYear(
+            definedBenefitInputAmount,
+            _,
+            preAccessDefinedContributionInputAmount,
+            postAccessDefinedContributionInputAmount,
+            taxYearSchemes,
+            _,
+            _,
+            period
+          ) =>
+        PaacTaxYear2016PreAlignment.InitialFlexiblyAccessedTaxYear(
+          definedBenefitInputAmount + taxYearSchemes.map(_.revisedPensionInputAmount).sum,
```

Comment:
Can you extract this summation logic to a method, as it has a business meaning that is lost in the noise of the case statements. What does this represent in customer terms? It is repeated widely in this class.